File: data/apm_data_sample.py
import os

import pandas as pd
from tqdm import tqdm

from analysis.data.helper_functions import input_data_path
from analysis.data.umapping import reduction_factor, unsupervised_umap_wrapper, embed_distances_with_umap
from phyloAutoEncoder import autoencode_pairwise_distances
number_of_apm_folds=10

def main():
    for tag in tqdm(range(1, number_of_apm_folds + 1)):
        dir_path = os.path.join(input_data_path, 'my_apm_data', 'ultrametric','APM', str(tag))
        unsupervised_umap_wrapper(dir_path)

        distances = pd.read_csv(os.path.join(dir_path, 'tree_distances.csv'), index_col=0)
        unsup_model, unsupervised_df = autoencode_pairwise_distances(distances, reduction_factor, dir_path, plot=True)
        unsupervised_df.to_csv(os.path.join(dir_path, 'unsupervised_autoencoded_phylogeny.csv'))

def with_full_tree():
    db_path = os.environ.get('KEWDROPBOXPATH')
    # Data from APM Traits
    binary_data = pd.read_csv(os.path.join(db_path, 'ApmTraits', 'apm_activity', 'outputs', 'compiled_extraction_apm_data.csv'))
    number_of_output_features = int(len(binary_data)*reduction_factor)

    repo_path = os.environ.get('KEWSCRATCHPATH')
    dir_path = os.path.join(repo_path, 'gentianales_trees', 'WCVP_12', 'Uphy', 'outputs',
                                        'Species')
    tree_path = os.path.join(dir_path, 'species_distances.csv')
    distances = pd.read_csv(tree_path, index_col=0)
    reduction_fraction = number_of_output_features/len(distances.columns)

    umap_embedding = embed_distances_with_umap(distances, reduction_fraction)
    umap_embedding.to_csv(os.path.join(dir_path, 'umap_unsupervised_embedding_full_tree.csv'))

    unsup_model, unsupervised_df = autoencode_pairwise_distances(distances, reduction_fraction, dir_path, plot=True)
    unsupervised_df.to_csv(os.path.join(dir_path, 'unsupervised_autoencoded_phylogeny_full_tree.csv'))


if __name__ == '__main__':
    main()
    with_full_tree()

File: data/apm_data_sample.R
## Create 10 fold CV to replicate ApmPP strategy

library(phytools)
library(ape)

source('helpful_phyl_methods.R')
source('helper_simulation_methods.R')
source('add_eigenvectors.R')
repo_path = Sys.getenv('KEWSCRATCHPATH')
species_tree = ape::read.tree(file.path(repo_path, 'gentianales_trees', 'WCVP_12', 'Uphy', 'outputs',
                                        'Species', 'Uphylomaker_species_tree.tre'))

db_path = Sys.getenv('KEWDROPBOXPATH')
# Data from APM Traits
binary_data = read.csv(file.path(db_path, 'ApmTraits', 'apm_activity','outputs','compiled_extraction_apm_data.csv'))
binary_data['accepted_species'] <- lapply(binary_data['accepted_species'], replace_space_with_underscore_in_name)

accepted_species = binary_data$accepted_species
useful_tree = subset_tree(species_tree,accepted_species)
tree_distances = ape::cophenetic.phylo(useful_tree)
binary_data = binary_data[binary_data$accepted_species %in% accepted_species,]
ground_truth = binary_data
# rownames(ground_truth) <- ground_truth$accepted_species
# ground_truth <- subset(ground_truth, select = -c(accepted_species))
# rownames(ground_truth) <- ground_truth$accepted_species
# ground_truth <- subset(ground_truth, select = -c(accepted_species))
# binary_sample = list(tree=sample_tree, FinalData= ground_truth, Dataframe=data.frame())
# ape::is.ultrametric(binary_sample$tree)
# output_simulation(file.path('my_apm_data'),binary_sample, binary_sample$tree,'binary', 1)


number_of_folds = 10
# This does unstratified sampling
skfolds = caret::createFolds(ground_truth[['APM.Activity']], k=number_of_folds)

output_apm_cv_data <- function(base_output_path, mcar_values){
    write.csv(tree_distances, file = file.path(base_output_path, 'tree_distances.csv'))
  ape::write.tree(useful_tree, file.path(base_output_path, 'tree.tre'))

  ## Save data
  this_sim_path = file.path(base_output_path, 'APM')
  dir.create(this_sim_path, recursive=TRUE)
  
  write.csv(ground_truth, file.path(this_sim_path, 'ground_truth.csv'),row.names = FALSE)
  ## Write missing values
  write.csv(mcar_values, file.path(this_sim_path, 'mcar_values.csv'),row.names = FALSE)
  saveRDS(mcar_values, file=file.path(this_sim_path, 'mcar_values.rds'))
  
  write.csv(data.frame(), file.path(this_sim_path, 'dataframe_params.csv'))


}


for (i in 1:number_of_folds) {
  fold_indices <- skfolds[[i]]
  cv_sample = ground_truth
  cv_sample[fold_indices, "APM.Activity"] <- NA
  output_apm_cv_data(file.path('my_apm_data', 'ultrametric', 'APM', i),cv_sample)
  decompose_tree(file.path('my_apm_data', 'ultrametric', 'APM', i))
}

for (i in 1:number_of_folds) {
  get_D_table(ground_truth, useful_tree,'APM.Activity',file.path('my_apm_data', 'ultrametric', 'APM', i, 'APM') )
}
